### DISCUSSION TITLE
Possible Bug in 5.10.0?

### DISCUSSION THREAD

**Discussion by achimste** [EDCD/EDMarketConnector#2120] [General]
Hi!

Since EDMC 5.10.0 I encounter a serious misbehaviour of EDMC together with TCE.

Description:
TCE is not able to read the market data when I dock on a station. TCE calls "EDMC.exe" to extract the market data and writes the data in the folder <TCE>/EDMC_Data. SO far so good. But TCE cannot read the data / is not able to incorporate the data into its databases.
I reinstalled EDMC 5.9 from a backup and that worked with TCE so latter one is not to blame. I found that EDMC.ex in 5.10.0 is missing the resource folder with two json files. These files are NOT installed with the 5.10.0 installer! I had to manually copy these from the EDMC release zip file.
Anyway, even with this "fix", TCE is still not working with 5.10, only with earlier version.

I already contacted Eventure of TCE. If anyone here has an idea, I'd be happy.

Greetings
Achim

**Comment by C1701D** [comment]
``ships.json`` and ``modules.json`` are only used by the outfitting functions, specifically the Coriolis and EDSY core plugins. The now deprecated ``ships.p`` and ``modules.p`` are also still included for backwards compatibility. That those two files don't show up in the ``resources`` folder is expected, they get added to the root path of the install location and a quick install test shows that both the new ``.json`` and the old ``.p`` files are present.

Unfortunately I'm unable to find any source code or proper code documentation for the Trade Computer Extension so I can't look into how exactly it's trying to get EDMC to generate the data, though I'm assuming it's using the ``-m`` flag to have ``EDMC.py`` generate a ``.csv`` file. If that's the case then EDMC should still generate log entries that might be able to give more details about what's happening or goign wrong. Could you look at the [normal](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#plain-log-file) and [debug logs](https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#debug-log-files) to either see if they note any errors or upload them for review?

(Additionally it would be preferred if you open an issue instead of a discussion.)

**Comment by C1701D** [comment]
That export function does rely on the ``companion.fixup()`` function which uses ``FDevIDs/rare_commodity.csv`` and ``FDevIDs/commodity.csv``. If those two files were missing the generated output file would be different but they are also both included with a fresh EDMC 5.10.0 install.

**Comment by Rixxan** [comment]
Created an issue to track. More information requested in #2124

**Comment by achimste** [comment]
I answered under issues with log files. I hope this helps.
[https://github.com/EDCD/EDMarketConnector/issues/2135](url)
